src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -0,0 +1,227 @@
+        private int _errorCount = 0;
+        private string _hostName;
+        private string _stampName;
+
+        public LinuxContainerMetricsPublisher(IEnvironment environment, IOptionsMonitor<StandbyOptions> standbyOptions, Action<LogLevel, string> logMetricsPublishEvent, HttpClient httpClient)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _logTraceEvent = logMetricsPublishEvent ?? throw new ArgumentNullException(nameof(logMetricsPublishEvent));
+            _memoryActivities = new BlockingCollection<MemoryActivity>(new ConcurrentQueue<MemoryActivity>(), boundedCapacity: _maxBufferSize);
+            _functionActivities = new BlockingCollection<FunctionActivity>(new ConcurrentQueue<FunctionActivity>(), boundedCapacity: _maxBufferSize);
+
+            _currentMemoryActivities = new ConcurrentQueue<MemoryActivity>();
+            _currentFunctionActivities = new ConcurrentQueue<FunctionActivity>();
+
+            var nodeIpAddress = _environment.GetEnvironmentVariable(EnvironmentSettingNames.NodeIpAddress);
+            _requestUri = string.Format(_requestUriFormat, nodeIpAddress);
+
+            _containerName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName);
+
+            if (_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange());
+            }
+            else
+            {
+                StartMetricsPublisher();
+            }
+        }
+
+        private void OnStandbyOptionsChange()
+        {
+            if (!_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                StartMetricsPublisher();
+            }
+        }
+
+        public void AddFunctionExecutionActivity(string functionName, string invocationId, int concurrency, string executionStage, bool success, long executionTimeSpan, DateTime eventTimeStamp)
+        {
+            FunctionActivity activity = new FunctionActivity
+            {
+                FunctionName = functionName,
+                InvocationId = invocationId,
+                Concurrency = concurrency,
+                ExecutionStage = executionStage,
+                IsSucceeded = success,
+                ExecutionTimeSpanInMs = executionTimeSpan,
+                EventTimeStamp = eventTimeStamp,
+                Tenant = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteStampDeploymentId)?.ToLowerInvariant()
+            };
+
+            if (!_functionActivities.TryAdd(activity))
+            {
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher: Buffer full. Dropping current batch of function activities"));
```

Comment:
After you declare an ILogger<LinuxContainerMetricsPublisher> ctor param for logging, it will log the Type name as the source, so you don't have to hard code in the trace event.